patches/server/0492-Fix-item-locations-dropped-from-campfires.patch

```diff
@@ -6,19 +6,21 @@ Subject: [PATCH] Fix item locations dropped from campfires
 Fixes #4259 by not flooring the blockposition among other weirdness
 
 diff --git a/src/main/java/net/minecraft/world/level/block/entity/CampfireBlockEntity.java b/src/main/java/net/minecraft/world/level/block/entity/CampfireBlockEntity.java
-index 9ef6b391268a99303aa853623d43c96073a13c9e..c6ae14009463b2d64acd4d3beb86d3233bcbae66 100644
+index 9ef6b391268a99303aa853623d43c96073a13c9e..6377e4bdada12e90522a0d3b3767efd2c45e7295 100644
 --- a/src/main/java/net/minecraft/world/level/block/entity/CampfireBlockEntity.java
 +++ b/src/main/java/net/minecraft/world/level/block/entity/CampfireBlockEntity.java
-@@ -79,7 +79,11 @@ public class CampfireBlockEntity extends BlockEntity implements Clearable {
+@@ -79,7 +79,13 @@ public class CampfireBlockEntity extends BlockEntity implements Clearable {
                          result = blockCookEvent.getResult();
                          itemstack1 = CraftItemStack.asNMSCopy(result);
                          // CraftBukkit end
 -                        Containers.dropItemStack(world, (double) pos.getX(), (double) pos.getY(), (double) pos.getZ(), itemstack1);
-+                    // Paper start
-+                    net.minecraft.world.entity.item.ItemEntity droppedItem = new net.minecraft.world.entity.item.ItemEntity(world, pos.getX() + 0.5D, pos.getY() + 0.5D, pos.getZ() + 0.5D, itemstack1.split(world.random.nextInt(21) + 10));
-+                    droppedItem.setDeltaMovement(world.random.nextGaussian() * 0.05D, world.random.nextGaussian() * 0.05D + 0.2D, world.random.nextGaussian() * 0.05D);
-+                    world.addFreshEntity(droppedItem);
-+                    // Paper end
++                        // Paper start
++                        while (!itemstack1.isEmpty()) {
++                            net.minecraft.world.entity.item.ItemEntity droppedItem = new net.minecraft.world.entity.item.ItemEntity(world, pos.getX() + 0.5D, pos.getY() + 0.5D, pos.getZ() + 0.5D, itemstack1.split(world.random.nextInt(21) + 10));
++                            droppedItem.setDeltaMovement(world.random.nextGaussian() * 0.05D, world.random.nextGaussian() * 0.05D + 0.2D, world.random.nextGaussian() * 0.05D);
```

Comment:
The velocity in recent version has changed here if it doesn't cause trouble with the original issue it should stay like vanilla

Reply:
My guess is 0.11485F

Reply:
In order to make the game more predictable some random events has been changed to use a triangle distribution instead of the Gaussian distribution to avoid extreme case (outside of -1 and 1). This transform the following calculation:
gauss(mean=0, deviation=1) * scale + base = triangle(mode, derivation)
The important part is:
derivation = 2.297 * scale.
mode = base
That been said i don't know why they take this number to calculate their derivation from the mean :/ probably to keep some consistency with the old behavior. If you apply those formulas to the code we get
float j = 0.05F;
world.random.triangle(0.0D, j * 2.297D)
But since mojang has saved 0.05 as a float in memory there's some rounding issue that appears (on top of how Java represent decimal number) 
so instead of 0.11485000000000001 we gets 0.11485000171139836 (those errors doesn't appear for the others triangle calculation).
If you want to keep the vanilla behavior (with this error) we needs to keep the float.
The exact value would be 0.11485 without any error.

Reply:
I found where the spread constant was, as for the error this seems to be the usual binary representation error caused by a float multiply by a double. The constant is RandomSource.GAUSSIAN_SPREAD_FACTOR (a double). So you can simplify this by:
double deviation = 0.05F * GAUSSIAN_SPREAD_FACTOR;
in the end this will produce the same number.

Reply:
Ok, updated to that. 

It's been so long, do we care that the x, y, z coords for the ItemEntity don't follow the exact calculations from `Containers#dropItemStack`? It's missing the stuff involving the item entity width.